src/messages.rs

```diff
@@ -3,14 +3,34 @@
 //! [RFC-001]: https://github.com/ZcashFoundation/redjubjub/blob/main/rfcs/0001-messages.md
 #![allow(dead_code)]
 
-use jubjub::{AffinePoint, Scalar};
-
+use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
 mod constants;
 mod validate;
 
+/// Define our own `Scalar` type instead of using jubjub::Scalar
+/// This is for serialization/deserialization purposes.
+///
+/// https://github.com/zkcrypto/jubjub/blob/main/src/lib.rs#L60
+/// https://github.com/zkcrypto/jubjub/blob/main/src/fr.rs#L20
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Scalar([u64; 4]);
+
+/// Define our own `AffinePoint` type instead of using jubjub::AffinePoint
+/// This is for serialization/deserialization purposes.
+///
+/// https://github.com/zkcrypto/jubjub/blob/main/src/lib.rs#L70-L73
+/// https://github.com/zkcrypto/jubjub/blob/main/src/lib.rs#L56
+/// https://github.com/zkcrypto/bls12_381/blob/main/src/scalar.rs#L20
+#[derive(Serialize, Deserialize, Debug)]
+pub struct AffinePoint {
+    u: Scalar,
+    v: Scalar,
+}
```

Comment:
There are multiple serialization issues here.

1. The derived `Scalar` byte order is incorrect on some platforms:
    - the serialization design specifies a 32-byte little-endian canonical representation
    - `[u64; 4]` is an array containing multi-byte integers
    - the derived serialization is little-endian on some platforms, and 64-bit big-endian on others
    - the serialization design specifically avoids multi-byte integers for this reason

2. The derived `AffinePoint` is 64 bytes:
    - the serialization design specifies a 32-byte little-endian canonical representation

Here's what the design says:
> We use `AffinePoint::to_bytes` and `AffinePoint::from_bytes` to get a 32-byte little-endian canonical representation. See https://github.com/zkcrypto/jubjub/blob/main/src/lib.rs#L443
> Similarly, `VerificationKey`s can be serialized using `<[u8; 32]>::from` and `VerificationKey::from`. See https://github.com/ZcashFoundation/redjubjub/blob/main/src/verification_key.rs#L86

```suggestion
/// Define our own `Scalar` type instead of using `jubjub::Scalar`.
///
/// The serialization design specifies that `Scalar` uses:
/// "a 32-byte little-endian canonical representation".
#[derive(Serialize, Deserialize, Debug)]
pub struct Scalar([u8; 32]);

/// Define our own `AffinePoint` type instead of using `jubjub::AffinePoint`.
///
/// The serialization design specifies that `AffinePoint` uses:
/// "a 32-byte little-endian canonical representation".
#[derive(Serialize, Deserialize, Debug)]
pub struct AffinePoint([u8; 32])
```

I've also made a corresponding fix to the design.

Reply:
(You can also tell `serde` to use specific types from other crates, and specific conversion methods. But this fix will work for now.)

Reply:
Yes i considered this but i was lazy. this is less code and easier.

Reply:
Great, thank you for this one. I merged in.